pkg/capabilities/consensus/ocr3/datafeeds/feeds_aggregator.go

```diff
@@ -47,25 +48,31 @@ var _ types.Aggregator = (*dataFeedsAggregator)(nil)
 // Metadata is a map of feedID -> (timestamp, price) representing onchain state (see DataFeedsOutcomeMetadata proto)
 func (a *dataFeedsAggregator) Aggregate(previousOutcome *types.AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value) (*types.AggregationOutcome, error) {
 	// find latest valid Mercury report for each feed ID
-	latestReportPerFeed := make(map[mercury.FeedID]mercury.Report)
+	latestReportPerFeed := make(map[mercury.FeedID]mercury.FeedReport)
 	for nodeID, nodeObservations := range observations {
 		// we only expect a single observation per node - new Mercury data
-		if len(nodeObservations) == 0 {
+		if len(nodeObservations) == 0 || nodeObservations[0] == nil {
 			a.lggr.Warnf("node %d contributed with empty observations", nodeID)
 			continue
 		}
 		if len(nodeObservations) > 1 {
```

Comment:
What are the circumstances when a node would contribute more than one observation? Do we look at those warnings? What's the expected course of action when this happens? Trying to understand why this is a `warn` and not an `error`/`critical`?

Reply:
A malicious node could send us any crap and we need to continue processing hoping to have a quorum. Here more than 1 obs can come from an invalid workflow too, which could be caught during validation (depends on how smart it is). Error logs might be too noisy. Soon I'm planning to also sprinkle the code with a bunch of Prom counters so we could easier detect anomalies coming from single nodes.